src/v/kafka/server/group_manager.cc

```diff
@@ -564,6 +578,42 @@ group_manager::leave_group(leave_group_request&& r) {
     }
 }
 
+ss::future<txn_offset_commit_response>
+group_manager::txn_offset_commit(txn_offset_commit_request&& r) {
+    if (!_catchup_lock.take_reader_lock()) {
+        vlog(klog.warn, "can't process a tx: coordinator_load_in_progress");
+        co_return txn_offset_commit_response(
+          r, error_code::coordinator_load_in_progress);
+    }
+
+    try {
+        auto error = validate_group_status(
+          r.ntp, r.data.group_id, offset_commit_api::key);
+        if (error != error_code::none) {
+            _catchup_lock.release_reader_lock();
+            co_return txn_offset_commit_response(r, error);
+        }
+
+        auto group = get_group(r.data.group_id);
+        if (!group) {
```

Comment:
should we mark `unlikely`?

Reply:
we don't do it with `offset_commit`, `offset_fetch`, `describe_group`, `delete_groups` so I don't thing tx methods are different